wondrous-app/components/Common/KanbanBoard/TaskColumn/index.tsx

```diff
@@ -172,29 +173,35 @@ function TaskColumn(props: ITaskColumn) {
         {(provided) => (
           <TaskListContainer ref={provided.innerRef} {...provided.droppableProps}>
             {cardsList?.length ? (
-              cardsList.map((card, index) => (
-                <Draggable key={card.id} draggableId={card.id} index={index} isDragDisabled={board?.isDragDisabled}>
-                  {(provided, snapshot) => (
-                    <div
-                      style={{
-                        width: '100%',
-                      }}
-                      {...provided.draggableProps}
-                      {...provided.dragHandleProps}
-                      ref={provided.innerRef}
-                      isDragging={snapshot.isDragging}
-                    >
-                      {card.type === ENTITIES_TYPES.MILESTONE && !card.isProposal ? (
-                        <Milestone>
+              cardsList.map((card, index) => {
+                const isDragDisabled = board?.isDragDisabled || taskHasPayment(card);
```

Comment:
nbd for now, but i think technically we should still be able to drag a task with payment within the completed section

Reply:
I agree, moveCard() from kanbanBoard is called on every card move, maybe we can check there if the task is paid and the new status is not completed => disable card move

other than this lgtm

Reply:
Yea, seems legit but it will cause "jumping" card back effect, I don't really like it. I have an idea how to make it better

Reply:
It definitely adds cool visual feedback for dnd